codegen-on-oss/codegen_on_oss/snapshot/context_snapshot.py

```
@@ -0,0 +1,213 @@
+        # Save locally if path provided
+        if local_path:
+            self._save_local(local_path)
+
+        # Save to S3 if bucket_store provided
+        if self.bucket_store:
+            self._save_to_s3()
+
+        logger.info(f"Created snapshot with ID: {self.snapshot_id}")
+        return self.snapshot_id
+
+    def _save_local(self, local_path: Union[str, Path]) -> None:
```

Ruff (0.8.2)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)